app/models/account.rb

```diff
@@ -0,0 +1,14 @@
+class Account < ApplicationRecord
+  belongs_to :group
+  belongs_to :owner, class_name: 'User', foreign_key: 'owner_id'
+
+  has_many :operations
+  has_many :account_credit_cards
+  has_many :account_debits
+  has_many :account_objectives
```

Comment:
The ideal of these relations is to be 1:1. One account, if credit card, has one `account_credit_card` which represents the details of the account.

The same of the other kinds. Important note: Only `money` has no account detail.

```suggestion
  has_one :account_credit_card
  has_one :account_debit
  has_one :account_objective
```